nodes/test/agent_rocketride/test_prune_wave_context.py

```
@@ -0,0 +1,264 @@
+    ``iGlobal`` context) so that the pruning logic can be tested in isolation.
+  - Only ``_context_window_waves`` and ``_wave_context_budget_chars`` are set;
+    no other driver state is needed by ``_prune_wave_context``.
+  - The method now returns a pruned *copy*; the master list passed in must
+    remain unchanged.  This invariant is explicitly verified by every test.
+
+Pattern mirrors: nodes/test/astra_db/test_convert_filter.py
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, List
```

Ruff (0.15.12)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)